crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -684,9 +701,7 @@ pub enum UnableToDecryptReason {
 
     /// Decryption failed because we're missing the megolm session that was used
     /// to encrypt the event.
-    ///
-    /// TODO: support withheld codes?
-    MissingMegolmSession,
+    MissingMegolmSession(Option<WithheldReason>),
```

Comment:
Could we instead move the existing `WithheldCode` into matrix-sdk-common, rather than inventing a new type that does basically the same thing?